csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -9,66 +9,61 @@
 
-        private readonly Encoding _payloadEncoding; // TODO: move to OutputStream
-
         /// <summary>Creates a new outgoing request frame with an OK reply status and a void return value.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        public static OutgoingResponseFrame Empty(Current current)
-            => new OutgoingResponseFrame(current, ReplyStatus.OK, ArraySegment<byte>.Empty);
+        /// <param name="encoding">The encoding for the frame payload.</param>
+        public static OutgoingResponseFrame Empty(Encoding encoding)
+            => new OutgoingResponseFrame(encoding, ReplyStatus.OK, ArraySegment<byte>.Empty);
 
         /// <summary>Creates a new outgoing response frame. It's a partial frame, and its payload needs to be
-        /// written later on with StartReturnValue/EndReturnValue.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        public OutgoingResponseFrame(Current current) : base(current.Adapter.Communicator, Ice1Definitions.Encoding)
+        /// written later on with WritePayload/SavePayload.</summary>
+        /// <param name="encoding">The encoding for the frame payload.</param>
+        public OutgoingResponseFrame(Encoding encoding) : base(encoding)
         {
-            RequestId = current.RequestId;
-            _payloadEncoding = current.Encoding;
-
-            WriteSpan(Ice1Definitions.ReplyHeader.AsSpan());
-            WriteInt(RequestId);
         }
 
         /// <summary>Creates a new outgoing response frame with the given reply status and payload.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
+        /// <param name="encoding">The encoding for the frame payload.</param>
         /// <param name="replyStatus">The reply status.</param>
         /// <param name="payload">The payload for this response frame.</param>
         // TODO: add parameter such as "bool assumeOwnership" once we add memory pooling.
-        public OutgoingResponseFrame(Current current, ReplyStatus replyStatus, ArraySegment<byte> payload)
-            : this(current)
+        public OutgoingResponseFrame(Encoding encoding, ReplyStatus replyStatus, ArraySegment<byte> payload)
+            : this(encoding)
         {
-            WriteByte((byte)replyStatus);
-
+            ReplyStatus = replyStatus;
+            var ostr = new OutputStream(Encoding, Data, PayloadStart);
+            ostr.WriteByte((byte)replyStatus);
             if (payload.Count == 0 && (replyStatus == ReplyStatus.OK || replyStatus == ReplyStatus.UserException))
```

Comment:
I don't think the payload can ever be empty for a user exception.

Reply:
Reworked this, the method no longer receives the reply status, the payload includes it, an empty payload implies `ReplyStatus.OK`